src/renderer/pipeline.rs

```diff
@@ -41,7 +41,7 @@ pub(crate) struct PipelineSharedState {
 pub(crate) struct Pipeline<T: RenderEngine> {
     hwnd: HWND,
     ctx: Context,
-    engine: T,
+    pub engine: T,
```

Comment:
I'd rather not make struct members public unless there is an explicit need to do so:
```suggestion
    engine: T,
```